packages/date-parser/src/index.js

```diff
@@ -61,9 +62,15 @@ const getParsedResultBoundaries = (parsedResults) => {
  * @param {Object} options
  * @param {Number} options.timezoneOffset Timezone offset in minutes
  * @param {OUTPUT_TYPES} options.output Type of the output dates
+ * @param {Number} weekStartDate Default weekStartDate is 1. 0 = Sunday, 1 = Monday, ..., 6 = Saturday
  * @return {ChronoNode.ParsedResult|Array}
  */
-export const parse = (str, ref, { timezoneOffset = 0, output = OUTPUT_TYPES.parsed_component } = {}) => {
+export const parse = (str, ref, { timezoneOffset = 0, output = OUTPUT_TYPES.parsed_component } = {}, weekStartDate = 1) => {
+  dayjs.locale({
```

Comment:
hmm this means that this function must not be async, otherwise race condition can happen.  
Can you look into `dayjs` and find out whether these is any other safer way to assign the locale?  
Otherwise, we should at least put a comment on top of this function: `we set the locale to dayjs globally, so we must look out for race condition`

Reply:
I think we could use `dayjs().locale()` instead of `dayjs.locale()`
https://day.js.org/docs/en/i18n/instance-locale

Reply:
instead of setting the wsd here. Another approach is to set the wsd right before doing week-related computations, for example, right at the beginning of `weekDay`, `lastX` and `xAgo`

Reply:
yeah, setting it locally would be safer. Can try doing that in `overrideDayjs`

Reply:
but need to make sure `overrideDayjs` only affects local instance of `chrono`

Reply:
ah, try setting locale locally in `momentFromStruct` á anh. All of our relative parsers use that function

Reply:
Updated `momentFromStruct`, `lastX` parsers, `weekday` passers and tests nha.